### ISSUE TITLE
Fail to retrieve event after 22nd

### ISSUE THREAD

**Issue by LiRiu** [hyperoracle/zkgraph-cli#31]
zkgraph cannot parse event which blocks[0].events.length is greater than 21.

### Env
macos version: Monterey 12.6
zkgraph-lib version: ^0.0.9-alpha.6 ( bug also exist in v1.0.0 )
nodejs version: v20.9.0

### Reproduce
reproduce as follow steps:
1. run command: `npm init zkgraph@alpha`
2. follow the instruction and initialize the zkgraph project
3. save the following code as mapping.ts:
```typescript
//@ts-ignore
import { Bytes, Block, Address } from "@hyperoracle/zkgraph-lib";
import { A } from "./a";

export function handleBlocks(blocks: Block[]): Bytes {
  const events = blocks[0].events;
  for(let i = 0; i < events.length; i++) {
    console.log("try to retrieve topic1 of event " + i.toString() + ": " + events[i].topic1.toHexString());
    const transfer = new A(events[i]);
    console.log("parse event " + i.toString() + " done. transfer event address: " + transfer.token.toHexString());
  }
  return Bytes.fromI32(1);
}
```

4. save the following code as a.ts:
```typescript
// @ts-ignore
import { Event, BigInt, Address } from "@hyperoracle/zkgraph-lib";

export class A {
  public token: Address;

  constructor(event: Event) {
    this.token = Address.fromBytes(event.topic1.slice(12,32));
  }
}

```
5. save the following code as zkgraph.yaml:
```yaml
specVersion: 0.0.2
apiVersion: 0.0.2
name: zkToken
description: "zkToken"
repository: https://github.com/hyperoracle/zkgraph
dataSources:
  - kind: ethereum
    network: mainnet
    event:
      - address: '0xc00e94cb662c3520282e6f5717214004a7f26888'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0x0000000000085d4780B73119b644AE5ecd22b376'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0x956F47F50A910163D8BF957Cf5846D573E7f87CA'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0x1985365e9f78359a9B6AD760e32412f4a445E862'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0xE41d2489571d322189246DaFA5ebDe1F4699F498'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0x6B3595068778DD592e39A122f4f5a5cF09C90fE2'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0x8E870D67F660D95d5be530380D0eC0bd388289E1'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0xdac17f958d2ee523a2206206994597c13d831ec7'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0x89d24A6b4CcB1B6fAA2625fE562bDD9a23260359'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0x514910771AF9Ca656af840dff83E8264EcF986CA'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0x0bc529c00C6401aEF6D220BE8C6Ea1667F6Ad93e'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0x0D8775F648430679A709E98d2b0Cb6250d2887EF'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0x6B175474E89094C44Da98b954EedeAC495271d0F'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984'
        events: 
          - "Transfer(address,address,uint256)"
      - address: '0x9f8F72aA9304c8B593d555F12eF6589cC3A579A2'
        events: 
          - "Transfer(address,address,uint256)"

mapping:
  language: wasm/assemblyscript
  file: ./mapping.ts
  handler: handleBlocks

dataDestinations:
  - kind: ethereum
    network: sepolia
    address: "0x0000000000000000000000000000000000000001"
```

6. run command: `npm run compile && npm run exec -- 18734585`

finally, we will meet a panic like this:
```
...
parse event 20 done. transfer event address: 0xf89d7b9c864f589bbf53a82105107622b35eaa40
try to retrieve topic1 of event 21: 0x300030003000300030003000300030003000300000000000000000001c000000
parse event 21 done. transfer event address: 0x300030003000300000000000000000001c000000
try to retrieve topic1 of event 22: 0x
wasm://wasm/51eb6a3a:1


RuntimeError: unreachable
    at wasm://wasm/51eb6a3a:wasm-function[17]:0x12d7
    at wasm://wasm/51eb6a3a:wasm-function[29]:0x26f9
    at asmain (/Users/licongfei/Documents/events22/node_modules/@hyperoracle/zkgraph-api/dist/index.cjs:2194:21)
    at executeOnInputs (/Users/licongfei/Documents/events22/node_modules/@hyperoracle/zkgraph-api/dist/index.cjs:2324:16)
    at async executeOnDataPrep (/Users/licongfei/Documents/events22/node_modules/@hyperoracle/zkgraph-api/dist/index.cjs:2309:10)
    at async Object.execute (/Users/licongfei/Documents/events22/node_modules/@hyperoracle/zkgraph-api/dist/index.cjs:2301:10)
    at async exec (/Users/licongfei/Documents/events22/node_modules/@hyperoracle/zkgraph-cli/dist/index.cjs:538:17)

Node.js v20.9.0
```
<img width="1023" alt="image" src="https://github.com/hyperoracle/zkgraph-cli/assets/103174898/742df6d7-b6ac-4240-9b57-90de6d1e7c07">

**Comment by LiRiu** [comment]
Strangely, the instantiation of type A in the code is a necessary condition for reproducing the bug.

**Comment by lightmanM** [comment]
tracked in case https://www.notion.so/hyperoracle/Events-are-limited-to-21-528b61065b714e44aa2855d16055e490

**Comment by LiRiu** [comment]
fixed in commit: e3392f2f005505df75368dec60c5428ce4f1c2be
